src/libstd/thread/mod.rs

```diff
@@ -1168,22 +1134,15 @@ impl Thread {
             EMPTY => return, // no one was waiting
             NOTIFIED => return, // already unparked
             PARKED => {} // gotta go wake someone up
-            _ => panic!("inconsistent state in unpark"),
+            n => panic!("inconsistent state in unpark: {}", n),
         }
```

Comment:
I'd have expected this all to go away. Which guarantee is the `park` from parking_lot lacking such that we need to still do some work here?

Reply:
This should just be `unreachable!` since we never set the state to anything other than `EMPTY`, `NOTIFIED` or `PARKED`.

Reply:
True. Replaced both panics with `unreachable!()`.